addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -114,6 +114,17 @@ const FormEditor = options.Class.extend({
     _getMark: function () {
         return this.$target[0].dataset.mark;
     },
+    /**
+     * Replace all `"` character by `&quot;` and all `'` character by `&apos;`.
+     * This is needed in order to be able to perform querySelector of this
+     * type: `querySelector(`[name="${name}"]`)`.
+     *
+     * @param {HTMLElement} name
+     */
+    _getQuotesEncodedName(name) {
+        return name.replaceAll(/"/g, character => `&quot;`)
```

Comment:
Didn't test but if it works, `$.escapeSelector` is surely better

Reply:
Unfortunately, `$.escapeSelector` leads to error. Here is a situation that illustrates it:
If `""` is set as the "Label" of a field, `setLabelText()` will execute and will do among other things:
```js
const valueEncoded = $.escapeSelector(value);
const inputEls = this.$target[0].querySelectorAll('.s_website_form_input');
inputEls.forEach(el => el.name = valueEncoded);
```
After this code, here are the values of the variables:
`value` : `""`
`valueEncoded` : `\\"\\"`
`inputEls[0].name` : `\\"\\"`
`inputEls[0]` : `<textarea ... name="\"\""  ... > </textarea>`
The problem is that `this.$target[0].querySelector(`.s_website_form_input[name="${value}"]`)` returns `null`. This will lead to error when trying to change the visibility dependency of fields (see `_setVisibilityDependency()`).

Reply:
Indeed, thank you for checking :+1: